File: 2025/day02/main.py
import re

t1 = 0
t2 = 0
for l in re.findall(r'\d+-\d+', input()):
    x, y = map(int, l.split('-'))
    for i in range(x, y + 1):
        s = str(i)
        k = len(s)
        if k % 2 == 0 and s[:k//2] * 2 == s:
            t1 += i
        for j in range(1, k):
            if k % j == 0 and s[:j] * (k // j) == s:
                t2 += i
                break
print(t1)
print(t2)

File: 2025/day02/src/main.rs
use std::io::{self};

fn main() {
    let mut ans1 = 0;
    let mut ans2 = 0;
    let input = {
        let mut s = String::new();
        io::stdin().read_line(&mut s).unwrap();
        s.trim_end().to_owned()
    };
    for interval in input.split(',') {
        let (lo_str, hi_str) = interval.split_once('-').unwrap();
        let lo = lo_str.parse::<u64>().unwrap();
        let hi = hi_str.parse::<u64>().unwrap();
        for i in lo..=hi {
            let i_str = i.to_string();
            let k = i_str.len();
            if k % 2 == 0 && i_str[..k / 2].repeat(2) == i_str {
                ans1 += i;
            }
            if (1..=k / 2).any(|j| k % j == 0 && i_str[..j].repeat(k / j) == i_str) {
                ans2 += i;
            }
        }
    }
    println!("{}\n{}", ans1, ans2);
}
